driver-embedded/src/main/com/mongodb/embedded/client/MongoDBCAPIHelper.java

```diff
@@ -165,4 +172,49 @@ private static void checkInitialized() {
 
     private MongoDBCAPIHelper() {
     }
+
+    /**
+     * Represents libmongodbcapi_init_params
+     */
+    public static class MongoDBCAPIInitParams extends Structure implements MongoDBCAPI.InitParams {
```

Comment:
The core representation of `libmongodbcapi_init_params`:

```
typedef struct {
    const char* yaml_config;
    uint64_t log_flags;
    libmongodbcapi_log_callback log_callback;
    void* log_user_data;
} libmongodbcapi_init_params;
```

Declaring it in the helper allows the inclusion of the `logCallback` while keeping this code protected.

JNA requires the fields of a `Structure` to be public hence the checkstyle and findbugs excludes.

Reply:
Nice.  

I get why this extends `Structure` but I'm not clear why it implements the empty `InitParams` interface.  In the [example](https://java-native-access.github.io/jna/4.2.1/overview-summary.html#structures) from the JNA docs there's no interface like that required.  Can you explain?

Reply:
This was a leftover from simplifications made to MongoCAPI, where previously the whole struct format for the init params and callback were declared. 

Using an interface represents the struct while not providing any implementation details. However, I've updated it to a `Structure` type in MongoCAPI which is more correct and achieves the same result.